src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,499 @@
+            $c->get('wp_bookings_ui/assets/require.js')
+        ), [], false, true);
+        wp_localize_script('rc-app-require', 'RC_APP_REQUIRE_FILES', [
+            'app' => $assetsUrlMap->get(
+                $c->get('wp_bookings_ui/assets/bookings/app.min.js')
+            )
+        ]);
+
+        /*
+         * All application components located here
+         */
+        wp_enqueue_script('rc-app', $assetsUrlMap->get(
+            $c->get('wp_bookings_ui/assets/bookings/main.js')
+        ), [], false, true);
+
+        /*
+         * Enqueue all styles from assets URL map
+         */
+        foreach ($c->get('wp_bookings_ui/assets/styles') as $styleId => $styleDependency) {
+            wp_enqueue_style('rc-app-' . $styleId, $assetsUrlMap->get($styleDependency));
+        }
+
+        $state = $this->_onBookingsPage() ? $this->_getBookingsAppState($c) : $this->_getServiceAppState();
+
+        wp_localize_script('rc-app', 'EDDBK_APP_STATE', $state);
+    }
+
+    /**
+     * Register hook on admin init which will register everything else.
+     *
+     * @param EventManagerInterface $eventManager
+     * @param TemplateManager $templateManager
+     */
+    protected function _adminInit($eventManager, $templateManager, $c)
+    {
+        $metaboxConfig = $c->get('wp_bookings_ui/metabox');
+        /*
+         * Add metabox with availabilities configuration to
+         * service's edit page.
+         */
+        add_meta_box(
+            $metaboxConfig->get('id'),
+            $this->__($metaboxConfig->get('title')),
+            function () use ($templateManager) {
+                echo $this->_renderMetabox($templateManager);
+            },
+            $metaboxConfig->get('post_type')
+        );
+
+        /*
+         * Add screen options on bookings management page.
+         */
+        add_filter('screen_settings', function ($settings, \WP_Screen $screen) use ($templateManager) {
+            if (!$this->_onBookingsPage())
+                return $settings;
+
+            return $this->_renderBookingsScreenOptions($templateManager);
+        }, 10, 2);
+
+        /*
+         * @todo: this is not working as expected (nothing happens).
+         */
+//        $this->_attach('screen_settings', function ($event) use ($templateManager) {
+//            if (!$this->_onBookingsPage())
+//                return $event->getParam(0);
+//
+//            return $this->_renderBookingsScreenOptions($templateManager);
+//        }, 10);
+    }
+
+    /**
+     * Register pages in the admin menu.
+     *
+     * @param TemplateManager $templateManager
+     * @param ContainerInterface $c
+     */
+    protected function _adminMenu($templateManager, $c)
+    {
+        $rootMenuConfig = $c->get('wp_bookings_ui/menu/root');
+        $settingsMenuConfig = $c->get('wp_bookings_ui/menu/settings');
+        $aboutMenuConfig = $c->get('wp_bookings_ui/menu/about');
+
+        $this->bookingsPageId = add_menu_page(

```

Comment:
Can we make sure that the booking page ID is never requested before it is set?

Reply:
@romalytvynenko, why do you need to ask WP for the page ID? Can you use something constant instead, like  `$menuSlug` (the 4th param)?

Reply:
I'll check...

Reply:
WP adds some stuff in front of slug:
`toplevel_page_eddbk-bookings` when slug is `eddbk-bookings`

Reply:
I understand. But why do you need it? Can you not rely on it, and rely on something that *you* are in control of?

Reply:
Got it, I'll check

Reply:
I can check it using:
`strpos(get_current_screen()->id, $menuSlug) !== false`

Or: 
`$_GET['page'] === $menuSlug`, but I know :(